### PR TITLE
kv, gossip: remove misc deprecated system config code 

### PR DESCRIPTION
**Pull request by RichardJCai**
Release note: None

### CODE REVIEW

File: pkg/testutils/localtestcluster/local_test_cluster.go [hunk 1]
```diff
@@ -227,12 +226,6 @@ func (ltc *LocalTestCluster) Start(t testing.TB, baseCtx *base.Config, initFacto
 		cfg.Settings,
 		nil,
 	)
-	cfg.SystemConfigProvider = systemconfigwatcher.New(
-		keys.SystemSQLCodec,
-		cfg.Clock,
-		rangeFeedFactory,
-		zonepb.DefaultZoneConfigRef(),
-	)
```
Comment (ajwerner) [inline pkg/testutils/localtestcluster/local_test_cluster.go:235]: should we be deleting this?

**Review by ajwerner — COMMENTED** [review]
A thing of beauty! Let's get somebody on KV to have a look. Maybe @irfansharif or @nvanbenschoten?

File: pkg/testutils/localtestcluster/local_test_cluster.go (hunk 1 above)
Comment (RichardJCai) [reply pkg/testutils/localtestcluster/local_test_cluster.go]: I think we can delete this but I'll look into it into a follow up.

**Review by kvoli — APPROVED** [review]
LGTM, +@nvanbenschoten

nit: I think there should be a commit message. Referencing #76279 for posterity.

File: pkg/kv/kvclient/kvcoord/txn_interceptor_committer_test.go [hunk 2]
```diff
@@ -243,9 +243,7 @@ func TestTxnCommitterStripsInFlightWrites(t *testing.T) {
 	ba.Requests = nil
 	etArgsWithTrigger := etArgs
 	etArgsWithTrigger.InternalCommitTrigger = &roachpb.InternalCommitTrigger{
-		ModifiedSpanTrigger: &roachpb.ModifiedSpanTrigger{
-			SystemConfigSpan: true,
-		},
+		ModifiedSpanTrigger: &roachpb.ModifiedSpanTrigger{},
```
Comment (irfansharif) [inline pkg/kv/kvclient/kvcoord/txn_interceptor_committer_test.go:246]: Do we need to to set this to anything at all?

File: pkg/kv/kvserver/queue_test.go [hunk 3]
```diff
@@ -690,24 +690,6 @@ func TestAcceptsUnsplitRanges(t *testing.T) {
...
-	var sysCfg *config.SystemConfig
-	testutils.SucceedsSoon(t, func() error {
-		sysCfg = s.cfg.Gossip.DeprecatedGetSystemConfig()
-		if sysCfg == nil {
-			return errors.New("system config not yet present")
-		}
-		return nil
-	})
-	neverSplitsDesc := neverSplits.Desc()
-	if sysCfg.NeedsSplit(ctx, neverSplitsDesc.StartKey, neverSplitsDesc.EndKey) {
```
Comment (irfansharif) [inline pkg/kv/kvserver/queue_test.go:703]: Down in KV we have an alternative for the component that determines whether a keyspan has hard split points between them, you should be be able to get yourself a spanconfig.StoreReader (GetConfReader?) backed by the span configs infra, and have this test use that instead. Ditto below; this test is a bit hollowed out without it.

File: pkg/kv/txn.go [hunk 4]
```diff
@@ -951,9 +951,7 @@ func endTxnReq(commit bool, deadline *hlc.Timestamp, hasTrigger bool) *endTxnReq
 	alloc.req.Deadline = deadline
 	if hasTrigger {
 		alloc.req.InternalCommitTrigger = &roachpb.InternalCommitTrigger{
-			ModifiedSpanTrigger: &roachpb.ModifiedSpanTrigger{
-				SystemConfigSpan: true,
-			},
+			ModifiedSpanTrigger: &roachpb.ModifiedSpanTrigger{},
```
Comment (irfansharif) [inline pkg/kv/txn.go:954]: Do we need any trigger whatsoever?

File: pkg/kv/kvserver/store.go [hunk 5]
```diff
@@ -2193,7 +2193,6 @@ func (s *Store) GetConfReader(ctx context.Context) (spanconfig.StoreReader, erro
 
 	if s.cfg.SpanConfigsDisabled ||
```
Comment (irfansharif) [inline pkg/kv/kvserver/store.go:2194]: Should we get rid of all these knobs now that we're no longer gossipping the system config span: SpanConfigsDisabled, spanconfigstore.EnabledSetting, s.TestingKnobs().UseSystemConfigSpanForQueues.

File: pkg/testutils/localtestcluster/local_test_cluster.go [hunk 6]
```diff
@@ -12,6 +12,7 @@ package localtestcluster
 
 import (
 	"context"
+	"github.com/cockroachdb/cockroach/pkg/server/systemconfigwatcher"
```
Comment (irfansharif) [inline pkg/testutils/localtestcluster/local_test_cluster.go:15]: Linter will complain here.

**Review by irfansharif — COMMENTED** [review]
I've paged a lot of this out, surprisingly, but the broad strokes LGTM. I'm curious what's left in order to get rid of all this code: https://github.com/cockroachdb/cockroach/blob/609825e62e991c7fd39d4f97c800951257fb8038/pkg/config/system.go#L76-L84.

File: pkg/kv/txn.go (hunk 4 above)
Comment (RichardJCai) [reply pkg/kv/txn.go]: Yeah this `hasTrigger` branch can go.

File: pkg/testutils/localtestcluster/local_test_cluster.go (hunk 6 above)
Comment (RichardJCai) [reply pkg/testutils/localtestcluster/local_test_cluster.go]: Done

File: pkg/kv/kvserver/queue_test.go (hunk 3 above)
Comment (RichardJCai) [reply pkg/kv/kvserver/queue_test.go:703]: Done

File: pkg/kv/kvclient/kvcoord/txn_interceptor_committer_test.go (hunk 2 above)
Comment (RichardJCai) [reply pkg/kv/kvclient/kvcoord/txn_interceptor_committer_test.go]: Looks like just `InternalCommitTrigger` is enough

File: pkg/testutils/localtestcluster/local_test_cluster.go (hunk 6 above)
Comment (RichardJCai) [reply pkg/testutils/localtestcluster/local_test_cluster.go]: Done

File: pkg/kv/kvclient/kvcoord/txn_interceptor_committer_test.go (hunk 2 above)
Comment (RichardJCai) [reply pkg/kv/kvclient/kvcoord/txn_interceptor_committer_test.go]: Nope

**Review by irfansharif — APPROVED** [review]
> or at least update `SystemConfigSpan` to contain just system.descriptor and
system.zones.

I'm not sure why we need an unsplittable span to contain those two tables specifically but happy to learn in your subsequent PRs. This is really great stuff, thanks for cleaning up the mess.

File: pkg/kv/kvserver/store.go (hunk 5 above)
Comment (RichardJCai) [reply pkg/kv/kvserver/store.go:2194]: I swear I replied but the message disappeared. 

I tried removing it but some tests were failing and it wasn't obvious to me why. I added two more TODOs to remove `SpanConfigsDisabled` and `spanconfigstore.EnabledSetting` and you already had one for `s.TestingKnobs().UseSystemConfigSpanForQueues`

**Comment by RichardJCai** [comment]
Thanks for reviewing!

bors r+

**Pull request merged**